pkg/securitycontextconstraints/sccadmission/admission.go

```diff
@@ -206,20 +206,14 @@ func (c *constraint) computeSecurityContext(ctx context.Context, a admission.Att
 		return nil, "", nil, admission.NewForbidden(a, fmt.Errorf("securitycontextconstraints.security.openshift.io required check failed oddly"))
 	}
 
-	constraints, err := sccmatching.NewDefaultSCCMatcher(c.sccLister, nil).FindApplicableSCCs(ctx, a.GetNamespace())
+	constraints, err := c.sccLister.List(labels.Everything())
 	if err != nil {
 		return nil, "", nil, admission.NewForbidden(a, err)
 	}
 	if len(constraints) == 0 {
-		sccs, err := c.sccLister.List(labels.Everything())
-		if err != nil {
-			return nil, "", nil, admission.NewForbidden(a, err)
-		}
-		if len(sccs) == 0 {
-			return nil, "", nil, admission.NewForbidden(a, fmt.Errorf("no SecurityContextConstraints found in cluster"))
-		}
 		return nil, "", nil, admission.NewForbidden(a, fmt.Errorf("no SecurityContextConstraints found in namespace %s", a.GetNamespace()))
```

Comment:
This error might be misleading in this case, as we're not really searching in the namespace but rather we list everything available to the cluster. Wouldn't this error (deleted above) be more appropriate?
```go
return nil, "", nil, admission.NewForbidden(a, fmt.Errorf("no SecurityContextConstraints found in cluster"))
```

Reply:
great suggestion, fixed 👍